pkg/plugin/plugin.go

```diff
@@ -0,0 +1,253 @@
+// SPDX-FileCopyrightText: Copyright The Lima Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package plugin
+
+import (
+	"cmp"
+	"context"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"slices"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/lima-vm/lima/v2/pkg/usrlocalsharelima"
+)
+
+type Plugin struct {
+	Name        string `json:"name"`
+	Path        string `json:"path"`
+	Description string `json:"description,omitempty"`
+}
+
+func DiscoverPlugins() ([]Plugin, error) {
+	var plugins []Plugin
+	seen := make(map[string]bool)
+
+	dirs := getPluginDirectories()
+
+	for _, dir := range dirs {
+		pluginsInDir := scanDirectory(dir)
+		for _, plugin := range pluginsInDir {
+			if !seen[plugin.Name] {
+				plugins = append(plugins, plugin)
+				seen[plugin.Name] = true
+			}
+		}
+	}
+
+	slices.SortFunc(plugins,
+		func(i, j Plugin) int {
+			return cmp.Compare(i.Name, j.Name)
+		})
+
+	return plugins, nil
+}
+
+func getPluginDirectories() []string {
+	var dirs []string
+
+	selfDirs := usrlocalsharelima.SelfDirs()
+	dirs = append(dirs, selfDirs...)
+
+	pathEnv := os.Getenv("PATH")
+	if pathEnv != "" {
+		pathDirs := filepath.SplitList(pathEnv)
+		dirs = append(dirs, pathDirs...)
+	}
+
+	if libexecDir, err := usrlocalsharelima.LibexecLima(); err == nil {
+		if _, err := os.Stat(libexecDir); err == nil {
+			dirs = append(dirs, libexecDir)
+		}
+	}
+
+	return dirs
+}
+
+// isWindowsExecutableExt checks if the given extension is a valid Windows executable extension
+// according to PATHEXT environment variable.
+func isWindowsExecutableExt(ext string) bool {
+	if runtime.GOOS != "windows" {
+		return false
+	}
+
+	pathExt := os.Getenv("PATHEXT")
+	if pathExt == "" {
+		pathExt = ".COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC;.CPL"
+	}
+
+	extensions := strings.Split(strings.ToUpper(pathExt), ";")
+	extUpper := strings.ToUpper(ext)
+
+	for _, validExt := range extensions {
+		if validExt == extUpper {
+			return true
+		}
+	}
+	return false
+}
+
+func scanDirectory(dir string) []Plugin {
+	var plugins []Plugin
+
+	entries, err := os.ReadDir(dir)
+	if err != nil {
+		logrus.Debugf("Plugin discovery: failed to scan directory %s: %v", dir, err)
+
+		return plugins
+	}
+
+	for _, entry := range entries {
+		if entry.IsDir() {
+			continue
+		}
+
+		name := entry.Name()
+		if !strings.HasPrefix(name, "limactl-") {
+			continue
+		}
+
+		pluginName := strings.TrimPrefix(name, "limactl-")
+
+		if strings.Contains(pluginName, ".") {
+			ext := filepath.Ext(name)
+			if !isWindowsExecutableExt(ext) {
+				continue
+			}
+
+			pluginName = strings.TrimSuffix(pluginName, ext)
+		}
```

Comment:
This logic is not correct because it treats names with multiple dots (like `limactl-hello.world.exe`) differently.

I think you should move the GOOS check here, and just strip Windows executable extensions on Windows, and leave the rest unchanged:

```suggestion
		if runtime.GOOS == "windows" {
			ext := filepath.Ext(pluginName)
			if isWindowsExecutableExt(ext) {
				pluginName = strings.TrimSuffix(pluginName, ext)
			}
		}
```